### PR TITLE
Cody aggregated ping additions.

### PR DESCRIPTION
**Pull request by nathan-downs**
## Test plan

Verify by running locally and checking ping structure.
Verify that payload is properly formed and submitted to pubsub topic.

### CODE REVIEW

File: .worktrees/backport-5.0
Comment (unknwon) [inline .worktrees/backport-5.0:1]: This file seems to be committed accdidentally?

File: cmd/frontend/internal/app/updatecheck/client.go [hunk 1]
```diff
@@ -145,6 +145,11 @@ func getUsersActiveTodayCount(ctx context.Context, db database.DB) (_ int, err e
 	return usagestats.GetUsersActiveTodayCount(ctx, db)
 }
 
+// func getCodyUsersActiveTodayCount(ctx context.Context, db database.DB) (_ int, err error) {
+// 	defer recordOperation("getCodyUsersActiveTodayCount")(&err)
+// 	return usagestats.GetCodyUsersActiveTodayCount(ctx, db)
+// }
```
Comment (unknwon) [inline cmd/frontend/internal/app/updatecheck/client.go:151]: Do you actually want to remove these lines? Same for other occurrences.

File: internal/database/event_logs.go [hunk 2]
```diff
@@ -1355,6 +1358,60 @@ GROUP BY name, current_week
 ORDER BY name;
 `
 
+func (l *eventLogStore) AggregatedCodyEvents(ctx context.Context, now time.Time) ([]types.CodyAggregatedEvent, error) {
```
Comment (unknwon) [inline internal/database/event_logs.go:1361]: Could you add some tests for this method in https://github.com/sourcegraph/sourcegraph/blob/nd/add-cody-pings/internal/database/event_logs_test.go?

**Review by unknwon — COMMENTED** [review]
No blocker from my side 😁 Have you double-checked with the analytics team whether we need to update our BigQuery schema first for new fields?

File: cmd/frontend/internal/app/updatecheck/client.go [hunk 3]
```diff
@@ -449,6 +471,12 @@ func limitedUpdateBody(ctx context.Context, logger log.Logger, db database.DB) (
 	}
 	r.ActiveToday = usersActiveTodayCount > 0
 
+	// 	codyUsersActiveTodayCount, err := getCodyUsersActiveTodayCount(ctx, db)
+	// 	if err != nil {
+	// 		logFunc("getCodyUsersActiveTodayCount failed", log.Error(err))
+	// 	}
+	// 	r.CodyActiveToday = codyUsersActiveTodayCount > 0
+
```
Comment (unknwon) [inline cmd/frontend/internal/app/updatecheck/client.go:479]:
```suggestion
```

I'm Click Engineer 😛

**Review by unknwon — APPROVED** [review]

**Comment by unknwon** [comment]
Thank you for the followups! Let's wait for CI.

**Comment by unknwon** [comment]
@nathan-downs - Looks like the test failed: https://buildkite.com/sourcegraph/sourcegraph/builds/216628#0187e1d5-4283-40ac-bddc-c1f357e7f0bc/340

File: cmd/frontend/internal/app/updatecheck/client.go [hunk 4]
```diff
@@ -352,6 +363,17 @@ func getAndMarshalCodeHostVersionsJSON(_ context.Context, _ database.DB) (_ json
 	return json.Marshal(v)
 }
 
+func getAndMarshalCodyUsageJSON(ctx context.Context, db database.DB) (_ json.RawMessage, err error) {
+	defer recordOperation("getAndMarshalCodyUsageJSON")(&err)
+
+	codyUsage, err := usagestats.GetAggregatedSearchStats(ctx, db) //TODO: fix
```
Comment (akalia25) [inline cmd/frontend/internal/app/updatecheck/client.go:369]: do you still need the 
`// TODO: fix` here?

File: cmd/frontend/internal/app/updatecheck/handler.go [hunk 5]
```diff
@@ -720,6 +733,53 @@ func reserializeSearchUsage(payload json.RawMessage) (json.RawMessage, error) {
 	return json.Marshal(singlePeriodUsage)
 }
 
+// reserializeSearchUsage will reserialize a code intel usage statistics
```
Comment (akalia25) [inline cmd/frontend/internal/app/updatecheck/handler.go:736]: nit: I think this comment should `// reserializeCodyUsage ...`

**Review by akalia25 — APPROVED** [review]
LGTM! 🚀

**Pull request merged**